src/core/ucc_progress_queue_mt.c

```diff
@@ -27,6 +29,10 @@ static void ucc_pq_locked_mt_enqueue(ucc_progress_queue_t *pq,
 {
     ucc_pq_mt_locked_t *pq_mt = ucc_derived_of(pq, ucc_pq_mt_locked_t);
 
+    if (UCC_HAS_TIMEOUT(task->args)) {
+        task->started = ucc_get_accurate_time();
```

Comment:
s/ucc_get_accurate_time/ucc_get_time ?

Reply:
why can't we set the start time coll_post instead if conditions in the progress func?